tests/e2e/csi_static_provisioning_file_basic.go

```diff
@@ -0,0 +1,239 @@
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		var datacenters []string
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		finder := find.NewFinder(e2eVSphere.Client.Client, false)
+		cfg, err := getConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters, ",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			defaultDatacenter, err = finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(defaultDatacenter)
+			defaultDatastore, err = getDatastoreByURL(ctx, datastoreURL, defaultDatacenter)
+			if err == nil {
+				break
+			}
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Datastore is not found in the datacenter list")
+	})
+
+	/*
+		This test verifies the static provisioning workflow.
+
+		Test Steps:
+		1. Create File share and get the file share id.
+		2. Create PV Spec with volumeID set to file share ID created in Step-1, and PersistentVolumeReclaimPolicy is set to Delete.
+		3. Create PVC with the storage request set to PV's storage capacity.
+		4. Wait for PV and PVC to bound.
+		5. Create a POD.
+		6. Verify volume is attached to the node and volume is accessible in the pod by creating a file inside volume.
+		7. Verify container volume metadata is present in CNS cache.
+		8. Delete POD.
+		9. Delete PVC.
+		10. Verify PV is deleted automatically.
+	*/
+	ginkgo.It("Verify basic static provisioning workflow", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		err := connectCns(ctx, &e2eVSphere)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating file share")
+		cnsCreateReq := cnstypes.CnsCreateVolume{
+			This:        cnsVolumeManagerInstance,
+			CreateSpecs: []cnstypes.CnsVolumeCreateSpec{*getFileShareCreateSpec(defaultDatastore.Reference())},
+		}
+		cnsCreateRes, err := cnsmethods.CnsCreateVolume(ctx, e2eVSphere.CnsClient.Client, &cnsCreateReq)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		task, err := object.NewTask(e2eVSphere.Client.Client, cnsCreateRes.Returnval), nil
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		taskInfo, err := cns.GetTaskInfo(ctx, task)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		fileShareVolumeID := taskResult.GetCnsVolumeOperationResult().VolumeId.Id
+
+		// Deleting the volume with deleteDisk set to false
+		ginkgo.By("Deleting the fileshare with deleteDisk set to false")
```

Comment:
Is this the only way we can create a file share? This process of creating and deleting a volume will first register the volume as a CNS volume and then remove it. This might not catch some failure cases which we are seeing in static provisioning.

cc: @divyenpatel

Reply:
This may not be the only way. We may have to write new govmomi methods otherwise.

@shalini-b / @divyenpatel can you share the scenarios which you mention above. I'll try to see if we can workaround them.

Reply:
I think @shalini-b is talking about the case to create PV statically. and then Delete PV and re-create PV with the same name. There is an issue that the 2nd PV creation does not register volume with CNS and simply returns a false message that registration for the volume is successful. I think we should add a test for this scenario.

Reply:
We can add that test with another PR.

Reply:
Since we are creating file share using CNS API and then de-registering it from CNS, we are good.

We have a couple of more options.
- We can assume fileshare is pre-created by some script and we can read it from the env variable.
- we can find vSAN fileshare go APIs and create file share using it, so we do not need to create and then de-register using CNS API.